### PR TITLE
refactor(dart/analyzer plugin): update to latest version of plugins

### PR DESCRIPTION
**Pull request by keertip**
@scheglov @yjbanov

### CODE REVIEW

File: modules_dart/analyzer_plugin/lib/tasks.dart [hunk 1]
```diff
@@ -5,5 +5,6 @@ import 'package:analyzer/task/model.dart';
 
 /// The analysis errors associated with a target.
 /// The value combines errors represented by multiple other results.
-final CompositeResultDescriptor<List<AnalysisError>> HTML_ERRORS =
-    new CompositeResultDescriptor<List<AnalysisError>>('ANGULAR_HTML_ERRORS');
+final ResultDescriptor<List<AnalysisError>> HTML_ERRORS =
```
Comment (scheglov) [inline modules_dart/analyzer_plugin/lib/tasks.dart]: The new way is to use ListResultDescriptor<AnalysisError> now.

File: modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart [hunk 2]
```diff
@@ -118,54 +120,33 @@ class _AbstractDartTaskTest {
...
   Map<ResultDescriptor<dynamic>, dynamic> outputs;
 
   CacheEntry getCacheEntry(AnalysisTarget target) {
-    return entryMap.putIfAbsent(target, () => new CacheEntry());
+    return entryMap.putIfAbsent(target, () => new CacheEntry(target));
   }
 
   void setUp() {
     emptySource = _newSource('/test.dart');
     // prepare AnalysisContext
```
Comment (scheglov) [inline modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart]: Remove this line?

File: modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart [hunk 3]
```diff
@@ -118,54 +120,33 @@ class _AbstractDartTaskTest {
   Source emptySource;
 
   DartSdk sdk = new MockSdk();
-  _MockContext context = new _MockContext();
+  //_MockContext context = new _MockContext();
```
Comment (scheglov) [inline modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart]: Remove this line?

File: modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart [hunk 4]
```diff
@@ -3,21 +3,17 @@ library angular2.src.analysis.analyzer_plugin.src.tasks_test;
 import 'package:analyzer/file_system/file_system.dart';
```
Comment (scheglov) [inline modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart]: The name of the file - tasks_test.server.spec.dart is strange.
It's not a test for DAS, and it's not spec :-)

File: modules_dart/analyzer_plugin/lib/tasks.dart (hunk 1 above)
Comment (keertip) [inline modules_dart/analyzer_plugin/lib/tasks.dart]: done

File: modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart (hunk 3 above)
Comment (keertip) [inline modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart]: done

File: modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart (hunk 4 above)
Comment (yjbanov) [inline modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart]: `spec` is the angular convention (tests are treated as "specifications" of functionality). `server` means that the test runs on the "server", i.e. outside the browser. The test runner uses these infixes to pick the tests to run, so it is a hard requirement to name your files in a way the test runner understands. `_test` is not needed.

Comment (scheglov) [inline modules_dart/analyzer_plugin/test/tasks_test.server.spec.dart]: Got it.
Thank you for explaining this!

**Comment by keertip** [comment]
build fails on dart stable - the code needs an analyzer version that is not supported for 1.11 stable. Waiting on the analyzer team to release a analyzer package that can be used with 1.11.

**Comment by keertip** [comment]
@yjbanov PTAL

**Comment by keertip** [comment]
@scheglov

**Comment by scheglov** [comment]
LGTM

**Comment by yjbanov** [comment]
@keertip @scheglov Quick note that no action is necessary on your part. We have an automated system for pushing PRs into master now (called mary-poppins).

**Pull request closed without merging**